[Feat] Dynamic data parameters automatically refresh the relevant dynamic filters (#1051)
mckinsey/vizro: vizro-core/changelog.d/20250305_151541_petar_pejovic_update_dynamic_filter_from_dfp.md

Author: Test

<!--
A new scriv changelog fragment.

Uncomment the section that is right (remove the HTML comment wrapper).
-->

<!--
### Highlights ✨

- A bullet item for the Highlights ✨ category with a link to the relevant PR at the end of your entry, e.g. Enable feature XXX. ([#1](https://github.com/mckinsey/vizro/pull/1))

-->
<!--
### Removed

- A bullet item for the Removed category with a link to the relevant PR at the end of your entry, e.g. Enable feature XXX. ([#1](https://github.com/mckinsey/vizro/pull/1))

-->
### Added

> - Automatically update dynamic `Filter` and all its targets when a dynamic data `Parameter` is changed. ([#1051](https://github.com/mckinsey/vizro/pull/1051))

Reviewer: ```suggestion
-  Update all dynamic `Filters` and their associated targets whenever a dynamic data `Parameter` is modified.([#1051](https://github.com/mckinsey/vizro/pull/1051))
```

Author: I replaces the `"all"` with the `"relevant"` word because only filters that are linked to the same targets as modified dynamic data Parameter are updated (and the filters targets too). Should I use "corresponding" (or some other word) instead of the "relevant"?   

```md
- Update relevant dynamic `Filters` and their associated targets whenever a dynamic data `Parameter` is modified. ([#1051](https://github.com/mckinsey/vizro/pull/1051))
```

> - Update relevant dynamic `Filters` and their associated targets whenever a dynamic data `Parameter` is modified. ([#1051](https://github.com/mckinsey/vizro/pull/1051))

Reviewer 2: ```suggestion
- Dynamic data parameters automatically refresh the relevant dynamic filters. ([#1051](https://github.com/mckinsey/vizro/pull/1051))
```
